[gatsby/www]: Fix ESLint warnings

## Problem

When running `yarn lint:code` on Gatsby's repository you'll be greeted with some ESLint warnings. It would be nice to fix those warnings!

```
$ eslint --ignore-path .gitignore --ignore-path .prettierignore --ext .js,.jsx .

/mnt/e/Lennart/Development/GitHub/gatsby/examples/image-processing/src/pages/index.js
  211:68  warning  HTML entities must be escaped  react/no-unescaped-entities
  211:76  warning  HTML entities must be escaped  react/no-unescaped-entities
  250:40  warning  HTML entities must be escaped  react/no-unescaped-entities
  250:44  warning  HTML entities must be escaped  react/no-unescaped-entities
  259:24  warning  HTML entities must be escaped  react/no-unescaped-entities
  259:32  warning  HTML entities must be escaped  react/no-unescaped-entities
  259:43  warning  HTML entities must be escaped  react/no-unescaped-entities
  259:51  warning  HTML entities must be escaped  react/no-unescaped-entities
  270:24  warning  HTML entities must be escaped  react/no-unescaped-entities
  270:32  warning  HTML entities must be escaped  react/no-unescaped-entities
  270:43  warning  HTML entities must be escaped  react/no-unescaped-entities
  270:51  warning  HTML entities must be escaped  react/no-unescaped-entities

/mnt/e/Lennart/Development/GitHub/gatsby/examples/using-contentful/src/pages/image-api.js
   16:21  warning  HTML entities must be escaped  react/no-unescaped-entities
   65:59  warning  HTML entities must be escaped  react/no-unescaped-entities
   68:15  warning  HTML entities must be escaped  react/no-unescaped-entities
   69:28  warning  HTML entities must be escaped  react/no-unescaped-entities
   69:36  warning  HTML entities must be escaped  react/no-unescaped-entities
  228:16  warning  HTML entities must be escaped  react/no-unescaped-entities

/mnt/e/Lennart/Development/GitHub/gatsby/examples/using-contentful/src/pages/index.js
  45:21  warning  HTML entities must be escaped  react/no-unescaped-entities
  53:27  warning  HTML entities must be escaped  react/no-unescaped-entities

/mnt/e/Lennart/Development/GitHub/gatsby/examples/using-css-modules/src/pages/index.js
  12:14  warning  HTML entities must be escaped  react/no-unescaped-entities

/mnt/e/Lennart/Development/GitHub/gatsby/examples/using-drupal/src/pages/index.js
  129:17  warning  Missing "key" prop for element in iterator  react/jsx-key
  150:34  warning  HTML entities must be escaped               react/no-unescaped-entities
  189:17  warning  Missing "key" prop for element in iterator  react/jsx-key

/mnt/e/Lennart/Development/GitHub/gatsby/examples/using-drupal/src/pages/recipes.js
  13:11  warning  Missing "key" prop for element in iterator  react/jsx-key

/mnt/e/Lennart/Development/GitHub/gatsby/examples/using-drupal/src/templates/recipe.js
  56:23  warning  HTML entities must be escaped               react/no-unescaped-entities
  62:55  warning  Missing "key" prop for element in iterator  react/jsx-key
  69:65  warning  Missing "key" prop for element in iterator  react/jsx-key

/mnt/e/Lennart/Development/GitHub/gatsby/examples/using-redirects/src/pages/index.js
  56:22  warning  HTML entities must be escaped  react/no-unescaped-entities
  56:37  warning  HTML entities must be escaped  react/no-unescaped-entities

/mnt/e/Lennart/Development/GitHub/gatsby/packages/gatsby/cache-dir/static-entry.js
  55:5  warning  Missing "key" prop for element in array  react/jsx-key

/mnt/e/Lennart/Development/GitHub/gatsby/www/src/components/evaluation-table.js
   20:9   warning  Missing "key" prop for element in array  react/jsx-key
  117:17  warning  Missing "key" prop for element in array  react/jsx-key
  118:17  warning  Missing "key" prop for element in array  react/jsx-key
  123:23  warning  Missing "key" prop for element in array  react/jsx-key
  127:23  warning  Missing "key" prop for element in array  react/jsx-key
  164:23  warning  Missing "key" prop for element in array  react/jsx-key

/mnt/e/Lennart/Development/GitHub/gatsby/www/src/pages/features.js
  46:5  warning  Missing "key" prop for element in array  react/jsx-key
  49:5  warning  Missing "key" prop for element in array  react/jsx-key
  52:5  warning  Missing "key" prop for element in array  react/jsx-key
  55:5  warning  Missing "key" prop for element in array  react/jsx-key
  58:5  warning  Missing "key" prop for element in array  react/jsx-key
  64:5  warning  Missing "key" prop for element in array  react/jsx-key
  67:5  warning  Missing "key" prop for element in array  react/jsx-key
  68:5  warning  Missing "key" prop for element in array  react/jsx-key
  69:5  warning  Missing "key" prop for element in array  react/jsx-key
  70:5  warning  Missing "key" prop for element in array  react/jsx-key

✖ 47 problems (0 errors, 47 warnings)
```

## Solution

Follow the [How to Contribute](https://www.gatsbyjs.org/docs/how-to-contribute/) guide. If you want to validate the fixes in the browser console you'll also need to run the `www` (gatsbyjs.org) site at some point.

To speed up the `www` developing you can pass an env variable for placeholder images. You'll also need to [add a GitHub token](https://github.com/gatsbyjs/gatsby/tree/master/www#environment-variables). In the end your `.env.development` file should contain:

```
GITHUB_API_TOKEN=your-github-token
GATSBY_SCREENSHOT_PLACEHOLDER=true
```

To validate that you fixed the warnings  run `yarn lint:code` in the root of the repo again.

--- a/examples/image-processing/src/pages/index.js
+++ b/examples/image-processing/src/pages/index.js
@@ -208,7 +208,7 @@ class Index extends React.Component {
           and a{` `}
           <code>base64</code>
           {` `}
-          image to use as a placeholder) you need to implement the "blur up"
+          image to use as a placeholder) you need to implement the {`"blur up"`}
           technique popularized by
           {` `}
           <a href="https://jmperezperez.com/medium-image-progressive-loading-placeholder/">
@@ -247,8 +247,8 @@ class Index extends React.Component {
 
         <p>
           The <code>toFormat</code> option lets you convert the source image to
-          another image format. We use "PNG" here to ensure that the duotoned
-          image does not show any JPG artifacts.
+          another image format. We use {`"PNG"`} here to ensure that the
+          duotoned image does not show any JPG artifacts.
         </p>
 
         <h3>
@@ -256,7 +256,8 @@ class Index extends React.Component {
             fluid(duotone:
             {` `}
             {`{ `}
-            highlight: "#f00e2e", shadow: "#192550" {`}`}, toFormat: PNG)
+            highlight: {`"#f00e2e"`}, shadow: {`"#192550"`} {`}`}, toFormat:
+            PNG)
           </small>
         </h3>
 
@@ -267,7 +268,7 @@ class Index extends React.Component {
             fluid(duotone:
             {` `}
             {`{ `}
-            highlight: "#0ec4f1", shadow: "#192550", opacity: 50 {`}`})
+            highlight: {`"#0ec4f1"`}, shadow: {`"#192550"`}, opacity: 50 {`}`})
           </small>
         </h3>
 

--- a/examples/using-contentful/src/pages/image-api.js
+++ b/examples/using-contentful/src/pages/image-api.js
@@ -13,7 +13,7 @@ const ImageAPI = props => {
         Gatsby offers rich integration with
         {` `}
         <a href="https://www.contentful.com/developers/docs/references/images-api/">
-          Contentful's Image API
+          {`Contentful's Image API`}
         </a>
       </p>
       <p>
@@ -62,11 +62,11 @@ const ImageAPI = props => {
       <p>
         If you make queries with <code>fixed</code> then Gatsby automatically
         generates images with 1x, 1.5x, 2x, and 3x versions so your images look
-        great on whatever screen resolution of device they're on.
+        great on whatever screen resolution of device {`they're`} on.
       </p>
       <p>
-        If you're on a retina class screen, notice how much sharper these images
-        are than the above "resized" images.
+        If {`you're`} on a retina class screen, notice how much sharper these
+        images are than the above {`"resized"`} images.
       </p>
       <p>
         You should prefer this operator over <code>resize</code>.
@@ -225,8 +225,8 @@ const ImageAPI = props => {
         WebP is currently only supported in
         {` `}
         <a href="https://caniuse.com/#feat=webp">Chrome and Oprah browsers</a>,
-        and you'll want to fall back to another format for other clients. When
-        this query is used with
+        and {`you'll`} want to fall back to another format for other clients.
+        When this query is used with
         {` `}
         <a href="https://www.gatsbyjs.org/packages/gatsby-image/">
           <code>gatsby-image</code>

--- a/examples/using-contentful/src/pages/index.js
+++ b/examples/using-contentful/src/pages/index.js
@@ -42,7 +42,7 @@ class IndexPage extends React.Component {
     return (
       <Layout>
         <div style={{ marginBottom: rhythm(2) }}>
-          <h2>Gatsby's integration with the Contentful Image API</h2>
+          <h2>{`Gatsby's`} integration with the Contentful Image API</h2>
           <Link to="/image-api/">See examples</Link>
           <br />
           <br />
@@ -50,8 +50,8 @@ class IndexPage extends React.Component {
           <h2>Localization</h2>
           <p>
             The <code>gatsby-source-contentful</code> plugin offers full support
-            for Contentful's localization features. Our sample space includes
-            products localized into both English and German.
+            for {`Contentful's`} localization features. Our sample space
+            includes products localized into both English and German.
           </p>
           <p>
             An entry and asset node are created for each locale following

--- a/examples/using-css-modules/src/pages/index.js
+++ b/examples/using-css-modules/src/pages/index.js
@@ -9,7 +9,8 @@ class IndexComponent extends React.Component {
       <div className={indexStyles.index}>
         <h1>Hello world</h1>
         <h2 className={indexStyles.subheader}>
-          You've arrived at the world renowned css modules & gatsby example site
+          {`You've`} arrived at the world renowned css modules & gatsby example
+          site
         </h2>
         <p>
           <Link className={indexStyles.link} to="/another-page/">

--- a/examples/using-drupal/src/pages/index.js
+++ b/examples/using-drupal/src/pages/index.js
@@ -125,13 +125,13 @@ class IndexPage extends React.Component {
               }}
             >
               <FirstPromoted recipe={topRecipe} />
-              {nextTwoPromotedRecipes.map((recipe, index) => (
+              {nextTwoPromotedRecipes.map(recipe => (
                 <PromotedCard
                   recipe={recipe}
                   square={true}
                   columns={4}
                   marginBottom={0}
-                  key={`promoted-card-${index}`}
+                  key={recipe.fields.slug}
                 />
               ))}
             </div>
@@ -186,11 +186,11 @@ class IndexPage extends React.Component {
                 flexWrap: `wrap`,
               }}
             >
-              {nextFourPromotedRecipes.map((recipe, index) => (
+              {nextFourPromotedRecipes.map(recipe => (
                 <PromotedCard
                   recipe={recipe}
                   columns={2}
-                  key={`promoted-card-${index}`}
+                  key={recipe.fields.slug}
                 />
               ))}
             </div>

--- a/examples/using-drupal/src/pages/index.js
+++ b/examples/using-drupal/src/pages/index.js
@@ -125,12 +125,13 @@ class IndexPage extends React.Component {
               }}
             >
               <FirstPromoted recipe={topRecipe} />
-              {nextTwoPromotedRecipes.map(recipe => (
+              {nextTwoPromotedRecipes.map((recipe, index) => (
                 <PromotedCard
                   recipe={recipe}
                   square={true}
                   columns={4}
                   marginBottom={0}
+                  key={`promoted-card-${index}`}
                 />
               ))}
             </div>
@@ -147,7 +148,7 @@ class IndexPage extends React.Component {
               }}
             >
               <div css={{ maxWidth: rhythm(15) }}>
-                <h2>In this month's edition</h2>
+                <h2>In this {`month's`} edition</h2>
                 <p>
                   Quisque vitae pulvinar arcu. Aliquam ac pellentesque erat, at
                   finibus massa.
@@ -185,8 +186,12 @@ class IndexPage extends React.Component {
                 flexWrap: `wrap`,
               }}
             >
-              {nextFourPromotedRecipes.map(recipe => (
-                <PromotedCard recipe={recipe} columns={2} />
+              {nextFourPromotedRecipes.map((recipe, index) => (
+                <PromotedCard
+                  recipe={recipe}
+                  columns={2}
+                  key={`promoted-card-${index}`}
+                />
               ))}
             </div>
           </Container>

--- a/examples/using-drupal/src/pages/recipes.js
+++ b/examples/using-drupal/src/pages/recipes.js
@@ -9,8 +9,8 @@ const AllRecipes = ({ data }) => (
     <Container>
       <h1>Recipes</h1>
       <ul>
-        {data.allRecipes.edges.map(({ node }, index) => (
-          <li key={index}>
+        {data.allRecipes.edges.map(({ node }) => (
+          <li key={node.fields.slug}>
             <Link to={node.fields.slug}>{node.title}</Link>
           </li>
         ))}

--- a/examples/using-drupal/src/pages/recipes.js
+++ b/examples/using-drupal/src/pages/recipes.js
@@ -9,8 +9,8 @@ const AllRecipes = ({ data }) => (
     <Container>
       <h1>Recipes</h1>
       <ul>
-        {data.allRecipes.edges.map(({ node }) => (
-          <li>
+        {data.allRecipes.edges.map(({ node }, index) => (
+          <li key={index}>
             <Link to={node.fields.slug}>{node.title}</Link>
           </li>
         ))}

--- a/examples/using-drupal/src/templates/recipe.js
+++ b/examples/using-drupal/src/templates/recipe.js
@@ -53,20 +53,24 @@ const RecipeTemplate = ({ data }) => (
           </div>
         </div>
         <div css={{ background: `white`, padding: rhythm(1.5) }}>
-          <h2>What you'll need and how to make this dish</h2>
+          <h2>What {`you'll`} need and how to make this dish</h2>
           <div css={{ display: `flex`, justifyContent: `space-between` }}>
             <div css={{ width: `calc(1/2*100% - (1 - 1/2) * ${rhythm(1.5)})` }}>
               <h3>Ingredients</h3>
               <ul>
                 {data.recipes.ingredients &&
-                  data.recipes.ingredients.map(ing => <li>{ing}</li>)}
+                  data.recipes.ingredients.map((ing, index) => (
+                    <li key={index}>{ing}</li>
+                  ))}
               </ul>
             </div>
             <div css={{ width: `calc(1/2*100% - (1 - 1/2) * ${rhythm(1.5)})` }}>
               <h3>Method</h3>
               <ul>
                 {data.recipes.instructions &&
-                  data.recipes.instructions.split(`,`).map(i => <li>{i}</li>)}
+                  data.recipes.instructions
+                    .split(`,`)
+                    .map(i => <li key={i}>{i}</li>)}
               </ul>
             </div>
           </div>

--- a/examples/using-redirects/src/pages/index.js
+++ b/examples/using-redirects/src/pages/index.js
@@ -53,7 +53,7 @@ const IndexPage = () => (
         to apply the <code>selected</code>
         &nbsp;class to links that match to the active path, and
         {` `}
-        <code>import './selected.css'</code>
+        <code>import {`'./selected.css'`}</code>
         &nbsp;in <code>/src/layouts/index.js</code>
         &nbsp;makes the <code>.selected {`{ color: #396; }`}</code> &nbsp;style
         declaration globally available to both&nbsp;pages.

--- a/packages/gatsby/cache-dir/static-entry.js
+++ b/packages/gatsby/cache-dir/static-entry.js
@@ -52,7 +52,11 @@ const createElement = React.createElement
 export default (pagePath, callback) => {
   let bodyHtml = ``
   let headComponents = [
-    <meta name="generator" content={`Gatsby ${gatsbyVersion}`} />,
+    <meta
+      name="generator"
+      content={`Gatsby ${gatsbyVersion}`}
+      key={`generator-${gatsbyVersion}`}
+    />,
   ]
   let htmlAttributes = {}
   let bodyAttributes = {}

--- a/www/src/components/evaluation-table.js
+++ b/www/src/components/evaluation-table.js
@@ -28,6 +28,7 @@ class EvaluationTable extends Component {
               background: colors.ui.bright,
             },
           }}
+          key={`info-icon-${words[words.length - 1]}`}
         >
           {` `}
           {`${words[words.length - 1]} `}
@@ -114,8 +115,11 @@ class EvaluationTable extends Component {
           {flatten(
             sections.map((section, s) =>
               [
-                <SectionTitle text={sectionHeaders[s]} />,
-                <SectionHeaderTop />,
+                <SectionTitle
+                  text={sectionHeaders[s]}
+                  key={`section-title-${s}`}
+                />,
+                <SectionHeaderTop key={`section-header-${s}`} />,
               ].concat(
                 flatten(
                   section.map((row, i) =>
@@ -123,8 +127,9 @@ class EvaluationTable extends Component {
                       <SectionHeaderBottom
                         display={row.node.Subcategory}
                         category={row.node.Subcategory}
+                        key={`section-header-bottom-${i}`}
                       />,
-                      <tr>
+                      <tr key={`first-row-${i}`}>
                         {headers.map((header, j) => (
                           <td
                             key={j}
@@ -165,6 +170,7 @@ class EvaluationTable extends Component {
                         style={{
                           display: showTooltip(s, i) ? `table-row` : `none`,
                         }}
+                        key={`second-row-${i}`}
                       >
                         <td
                           css={{

--- a/www/src/pages/features.js
+++ b/www/src/pages/features.js
@@ -43,31 +43,54 @@ const LegendTable = () => {
   }
 
   const balls = [
-    <div css={legendBallCellStyle}>
+    <div css={legendBallCellStyle} key={`${legendBallCellStyle}-1`}>
       <h4 style={{ margin: 0 }}>Icon</h4>
     </div>,
-    <div css={legendBallCellStyle}>
+    <div css={legendBallCellStyle} key={`${legendBallCellStyle}-2`}>
       <EvaluationCell num="3" style={legendBallStyle} />
     </div>,
-    <div css={legendBallCellStyle}>
+    <div css={legendBallCellStyle} key={`${legendBallCellStyle}-3`}>
       <EvaluationCell num="2" style={legendBallStyle} />
     </div>,
-    <div css={legendBallCellStyle}>
+    <div css={legendBallCellStyle} key={`${legendBallCellStyle}-4`}>
       <EvaluationCell num="1" style={legendBallStyle} />
     </div>,
-    <div css={legendBallCellStyle}>
+    <div css={legendBallCellStyle} key={`${legendBallCellStyle}-5`}>
       <EvaluationCell num="0" style={legendBallStyle} />
     </div>,
   ]
 
   const legendText = [
-    <div css={legendExplanationCellStyle}>
+    <div
+      css={legendExplanationCellStyle}
+      key={`${legendExplanationCellStyle}-1`}
+    >
       <h5 style={{ margin: 0 }}>Feature Availability</h5>
     </div>,
-    <div css={legendExplanationCellStyle}>Out of the box</div>,
-    <div css={legendExplanationCellStyle}>Plugins available</div>,
-    <div css={legendExplanationCellStyle}>Needs customization</div>,
-    <div css={legendExplanationCellStyle}>Not possible</div>,
+    <div
+      css={legendExplanationCellStyle}
+      key={`${legendExplanationCellStyle}-2`}
+    >
+      Out of the box
+    </div>,
+    <div
+      css={legendExplanationCellStyle}
+      key={`${legendExplanationCellStyle}-3`}
+    >
+      Plugins available
+    </div>,
+    <div
+      css={legendExplanationCellStyle}
+      key={`${legendExplanationCellStyle}-4`}
+    >
+      Needs customization
+    </div>,
+    <div
+      css={legendExplanationCellStyle}
+      key={`${legendExplanationCellStyle}-5`}
+    >
+      Not possible
+    </div>,
   ]
 
   return (

